log/identifier.go

```diff
@@ -24,5 +24,10 @@ import (
 // for this log at distributors, and that will be used to feed
 // checkpoints to witnesses.
 func ID(origin string, key []byte) string {
-	return fmt.Sprintf("%x", sha256.Sum256(key))
+	// Hash both the origin and key, and then hash them together
+	// to create the final ID.
+	oh := sha256.Sum256([]byte(origin))
```

Comment:
This could also be something like below to avoid hashing 3 times:
```go
s := sha256.New()
s.Write([]byte("o:"))
s.Write([]byte(origin))
s.Write([]byte("\nk:"))
s.Write(key)
return fmt.Sprintf("%x", s.Sum(nil))
```

Reply:
Done.